horace_core/sqw/axes_blocks/@AxesBlockBase/private/pbin_parse_.m

```diff
@@ -0,0 +1,107 @@
+function [range,nbin,ok,mess]=pbin_parse_(~,p,p_defines_bin_centers,range_limits)
+% Check form of the bin descriptions and return bin boundaries
+% and number of bins for axes block built from the bin descriptors provided
+% as input.
+%
+%   >> [range,nbin]=pbin_parse(obj,p,p_defines_bin_centers,range_limits)
+%
+% Input:
+% ------
+%   p  --   Bin description
+%           - [pcent_lo,pstep,pcent_hi] (pcent_lo<=pcent_hi; pstep>0)
+%           - [pint_lo,pint_hi]         (pint_lo<=pint_hi)
+%  p_defines_bin_centers
+%     --   4-elements logical array, containing true if 3-component bin
+%          parameters define bin centers (default) or false if bin edges
+%          (exotic case)
+%  range_limits
+%      --  limits for ranges (min_range,max_range) this bins can have.
+%          For linear axis the bin limits are -inf,inf, but for spherical
+%          or cylindrical bins they have limited value for angles or radial
+%          components of momentum.
+%
+%
+% Output:
+% -------
+%   range  --  The min/max values of the range, covered by axis number i
+%              in selected direction.
+%   nbin   --  number of bins, the range is divided into (from 1(integration axis)
+%              to number of bins in projection axis)
+%
+ok = true;
+mess = '';
+if isempty(p)
+    range=[0;0];
+    nbin = 1;
+elseif isnumeric(p)
+    if numel(p)==1
+        % Scalar pbin ==> zero thickness integration? Useless. Current algorithm always leads to empty cut.
+        % May be left for a future, for doing interpolated 0-thin cuts on dnd objects?
```

Comment:
As it currently stands, single pbin value never comes here in reality.  You initially parse cut inputs, which converts single value into triplet. 

Then you parce axes block from these triplets. This may change for different cut types but we have not decided. Added the ticket whcih may need this feature.